ledger/apptxn_test.go

```diff
@@ -37,11 +48,158 @@ func main(source string) string {
        end: int 1`, source)
 }
 
+// newTestLedger creates a in memory Ledger that is as realistic as
+// possible.  It has Rewards and FeeSink properly configured.
+func newTestLedger(t testing.TB, balances bookkeeping.GenesisBalances) *Ledger {
```

Comment:
looks like this also need to go into `testing` since these functions are generic and not app-specific

Reply:
We can't do that due to circular dependencies between the "testing" and the "ledger".

Reply:
looks like the only problem is `*Ledger* at 
```golang
type testingEvaluator struct {
	*internal.BlockEvaluator
	ledger *Ledger
}
```
and `ledger. genesisProto`, `ledger.lookup` usages. The former is implemented as `GenesisProto()` in #2984, the latter most likely is `Lookup*` function. So defining a new `LedgerForTesting` (well, I know) interface might make the trick.